Backlog of fixes

FerretDave opened the pull request:
a variety of minor code fixes and various DESC cleanup.
Now submitted as individual branches, my repo has been rebuilt and I'm now using the command line

BahamutDragon commented on data/35e/wizards_of_the_coast/campaign_settings/forgotten_realms/races_of_faerun/rac_templates.lst:
```diff
@@ -52,4 +52,7 @@ Yuan-Ti (Tainted One)
-Reincarnated into Mountain Orc	KEY:Reincarnated ~ Mountain Orc	VISIBLE:EXPORT	TYPE:Reincarnation	SIZE:M	BONUS:STAT|STR|4|TYPE=Racial	BONUS:VAR|DarkvisionRange|60|TYPE=Base	ABILITY:Special Ability|AUTOMATIC|Orc ~ Light Sensitivity|Orc ~ Orc Blood|Remove Race Traits	MOVE:Walk,30
+# Orc Blood doesnt exist,only Half-Orc blood !
+#Orc ~ Orc Blood
+Reincarnated into Mountain Orc	KEY:Reincarnated ~ Mountain Orc	VISIBLE:EXPORT	TYPE:Reincarnation	SIZE:M	BONUS:STAT|STR|4|TYPE=Racial	BONUS:VAR|DarkvisionRange|60|TYPE=Base	ABILITY:Special Ability|AUTOMATIC|Orc ~ Light Sensitivity| |Remove Race Traits	MOVE:Walk,30
```
This should be throwing an error.   |Orc ~ Light Sensitivity| |Remove Race Traits. There should not be a '| |' in there.

BahamutDragon commented on data/35e/wizards_of_the_coast/campaign_settings/forgotten_realms/unapproachable_east/una_abilities.lst:
```diff
@@ -39,7 +39,8 @@ Runescarring										CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary
 Extra Rage			KEY:Runescarred Berserker ~ Extra Rage		CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary			DEFINE:RageTimes|0	DEFINE:RunescarExtraRage|0	DEFINE:RunescarExtraRageLVL|0	DESC:A runescarred berserker gains additional daily uses of the barbarian rage ability. At 2nd level, she gains one additional rage per day; at 5th level, she gains a second ...
-Ritual Scarring		KEY:Runescarred Berserker ~ Ritual Scarring	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary																	DESC:Through frequent disfiguration of her own skin, a runescarred berserker gains a +1 natural armor bonus to AC at 3rd level. This increases to +2 at 6th level, and +3 at 9th level.																																												 ...
-Spawn of the Frost	KEY:Runescarred Berserker ~ Spawn of the Frost	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Supernatural																	DESC:Inured to the bitter cold of Rashemen's winters, a runescarred berserker gains cold resistance 5.																																																																																																																									 ...
+Ritual Scarring		KEY:Runescarred Berserker ~ Ritual Scarring	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary																	DESC:Through frequent disfiguration of her own skin, a runescarred berserker gains a +1 natural armor bonus to AC at 3rd level. This increases to +2 at 6th level, and +3 at 9th level.																																												 ...
```
Why separate statements? This can be simplified with multiple if statements inside or using the native comparison.

BahamutDragon commented on data/35e/wizards_of_the_coast/campaign_settings/forgotten_realms/unapproachable_east/una_equip.lst:
```diff
@@ -57,30 +57,48 @@ Silver Sign	TYPE:Magic.Wondrous.Sign.Consumable	MODS:REQUIRED		COST:5000	WT:0.01
 # 1st level
 Runescar of Cure Moderate Wounds	SPROP:Cure 2d8+ % HP|(min((classlevel("Runescarred Berserker")),30))	TYPE:Magic.Consumable	PRESPELL:1,Cure Moderate Wounds	PRECLASS:1,Runescarred Berserker=1
+#DG todo - cannot add ASPECT to equipment, JIRA'd, may workaround by having equipment add an ability that then adds the aspect...
+#ASPECT:CombatBonus|+% luck bonus to hit/damage for 1 minute (Runescar of Divine Favour)|min(min((classlevel("Runescarred Berserker"))/3,1),3)
```
Why are we electing to have ASPECT comments sitting on equipment? Either grant the ability with the ASPECT tag, swap out for a QUALITY tag, or figure out how to implement the INFO tag since that seems to be the eventual direction with text output tags.

BahamutDragon commented on data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst:
```diff
@@ -152,20 +152,24 @@ Feat Bonus		KEY:Human ~ Bonus Feat	CATEGORY:Special Ability	TYPE:Race Trait.Huma
-Stonecunning		KEY:Dwarf ~ Stonecunning		CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY	DEFINE:StonecunningSearchBonus|0	DESC:This ability grants a dwarf a +%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings ...
-Stability			KEY:Dwarf ~ Stability			CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY							DESC:+4 bonus on ability checks made to resist being bull rushed or tripped when standing on the ground (but not when climbing, flying, riding, or otherwise not standing firmly on the ground).																																																																									 ...
+Stonecunning		KEY:Dwarf ~ Stonecunning		CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY	DEFINE:StonecunningSearchBonus|0	DESC:This ability grants a dwarf a +%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings ...
+#ASPECT:SkillBonus|+%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings, and the like. Something that isn't stone but that is disguised as stone also counts as unusual stonework. A dwarf who merely comes within 10 feet of unusual stonework can make a S ...
```
Why is the ASPECT commented out?

BahamutDragon commented on data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst:
```diff
@@ -270,7 +280,8 @@ Retain Dex to AC ~ Flat-footed												CATEGORY:Special Ability	TYPE:Internal
 # Ability Name		Category of Ability		Type												Description																																																																																																									Ability										Modify VAR
 Uncanny Dodge		CATEGORY:Special Ability	TYPE:Class Feature.SpecialQuality.Extraordinary.Uncanny Dodge					DESC:Character retains their Dexterity bonus to AC (if any) even if caught flat-footed or struck by an invisible attacker. However, they still loses his Dexterity bonus to AC if immobilized.																																																																											ABILITY:Special Ability|AUTO ...
-Improved Uncanny Dodge	CATEGORY:Special Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class Feature.SpecialQuality.Improved Uncanny Dodge	DESC:Character can no longer be flanked; you can react to opponents on opposite sides of you as easily as you can react to a single attacker. This defense denies a rogue the ability to sneak attack the character by flanking him, unless the attacker has at lea ...
+Improved Uncanny Dodge	CATEGORY:Special Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class Feature.SpecialQuality.Improved Uncanny Dodge	DESC:You can no longer be flanked; you can react to opponents on opposite sides of you as easily as you can react to a single attacker. This defense denies a rogue the ability to sneak attack you by flanking you, unless the attacker has at least four more rog ...
```
Why are we setting the numbers here incorrectly?

BahamutDragon commented on data/35e/wizards_of_the_coast/core/players_handbook/ph_equip.lst:
```diff
@@ -53,11 +53,11 @@ Touch Attack (Ray)								PROFICIENCY:WEAPON|Spells (Ray)			TYPE:Weapon.Ranged.U
 # Equipment Name			Unique Key					Required Weapon Proficiency				Type																	Alt Type						Contains													Cost		Weight	Slot Needed	Base Quantity	Crit Mult	Crit Range	Alt Crit Mult	Alt Crit Range	Damage	Alt Damage		Modifier			Alt EQModifier	Wield Category	Requires Modification?	Range		Reach		Reach Multiplier	Size		Source Page				Bonus Equip Mods																		Weapon prop. bonus		 ...
 Arrow																			TYPE:Ammunition.Standard.Arrow.Individual																																COST:0.05	WT:0.15			BASEQTY:1																	EQMOD:STEEL.WOOD								MODS:YES												SOURCEPAGE:p.114																																																			SPROP:An arrow used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls) and deals damage as a dagger of it's s ...
-Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt (Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A crossbow bolt used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls)and deals damage as a d ...
+Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt (Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A crossbow bolt used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls)and deals damage as a d ...
```
Why are you removing the escape character?

BahamutDragon commented on data/35e/wizards_of_the_coast/core/universal_rules/ur_abilities.lst:
```diff
@@ -917,11 +917,12 @@ Resistance to Sonic		CATEGORY:Special Ability	TYPE:ResistanceSelection	VISIBLE:N
-Resistance to Electricity Output	OUTPUTNAME:Resistance to Electricity	CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Electricity damage each time you take Electricity damage|ElectricityResistanceBonus	ASPECT:Resistance|Electricity	ASPECT:Ability Benefit|%1|ElectricityResistanceBonus
-Resistance to Sonic Output		OUTPUTNAME:Resistance to Sonic		CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Sonic damage each time you take Sonic damage|SonicResistanceBonus				ASPECT:Resistance|Sonic	ASPECT:Ability Benefit|%1|SonicResistanceBonus
+Resistance to Acid Output		OUTPUTNAME:Resistance to Acid			CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Acid damage each time you take Acid damage|AcidResistanceBonus					ASPECT:SaveBonus|Acid Resistance %1|AcidResistanceBonus	ASPECT:Resistance|Acid	ASPECT:Ability Benefit|%1|AcidResistanceBonus
+# DG - not seeing the original ASPECT features output anywhere, so adding additional
```
Just how many times do you need a resistance displayed? I think three times is a bit excessive for the PDF export. 

You might want to add the same functions of the pdf to the htm export for a consistent appearance.

BahamutDragon commented on data/35e/wizards_of_the_coast/supplement/expanded_psionics_handbook/xph_equip.lst:
```diff
@@ -1,129 +1,131 @@
-Psicrown (Great Dominator)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:45000	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 750; Manifester Level - 15; Powers - Charm (Psionic), Dominate (Psionic), Suggestion (Psionic)
-Psicrown (Temporal Juggler)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:95625	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 850; Manifester Level - 17; Powers - Temporal Acceleration, Time Hop, Timeless Body
-Psicrown (Traveler)								OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:80156	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 750; Manifester Level - 15; Powers - Astral Traveler, Dimension Door (Psionic), Teleport (Greater (Psionic)), Teleport (Psionic ...
+Psicrown (Astral Legion)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:47250	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:16																																																																						SPROP:Power Points - 900; Manifester Level - 18; Powers - Astral Construct
```
SOURCEPAGE:16? Does not match the surrounding three digit formats

BahamutDragon commented:
I think you are confusing terminology. You have Commits, not branches, within a single branch. Each commit may be a separate issue, but it is packaged in the same branch. A branch is meant to be a focus point for a single set of changes that are related. To simplify the concept, a branch is a box where you store similar items. Barbarian Alternative Class Features would go in a box, adding aspects would be another branch/box, adding an eqmod to handle consumables would be another branch/box. All of the boxes are separate from each other. Then each box is submitted alone for review, and is accepted or rejected based upon the content of the box. When you do mass PRs, such as this one, you have a big box with completely unrelated things jumbled in. Like clothes, toys, and tools all thrown together.  On top of that, you really strain my ability to do a proper review. Every other file has a problem, but after 12 or 14 files, it becomes harder to give a good review. Especially when the changes are a jumble and inconsistent.

Taken individually, these changes are positive things I would like to include. They need to be cleaned up, and they need to be separated for an easier review.

FerretDave commented:
Now I'm tearing my hair out!I *did* each change as a separate branch!!!‎As per my git command notes, I created a branch, checked it out, and ‎committed a few files (with related changes) to it. Did a git push and repeat.Then I opened the PR - which I thought covered all of the branches i'd submitted... this PR should have a doszen or so branches, each commit was *aslo* a seperate branch... i'll take another look.                                                                                                                                                                                                   CheersDave     oxxxxxxx{;;;;;;;;;;;;;;;;;;;;;;;;;;;;;>                                                                                                                                                                                                                From: BahamutSent: Friday, 14 October 2016 05:04To: BahamutDragon/pcgenReply To: BahamutDragon/pcgenCc: Dave; AuthorSubject: Re: [BahamutDragon/pcgen] Backlog of fixes (#588)I think you are confusing terminology. You have Commits, not branches, within a single branch. Each commit may be a separate issue, but it is packaged in the same branch. A branch is meant to be a focus point for a single set of changes that are related. To simplify the concept, a branch is a box where you store similar items. Barbarian Alternative Class Features would go in a box, adding aspects would be another branch/box, adding an eqmod to handle consumables would be another branch/box. All of the boxes are separate from each other. Then each box is submitted alone for review, and is accepted or rejected based upon the content of the box. When you do mass PRs, such as this one, you have a big box with completely unrelated things jumbled in. Like clothes, toys, and tools all thrown together.  On top of that, you really strain my ability to do a proper review. Every other file has a problem, but after 12 or 14 files, it becomes harder to give a good review. Especially when the changes are a jumble and inconsistent.

Taken individually, these changes are positive things I would like to include. They need to be cleaned up, and they need to be separated for an easier review. 

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/BahamutDragon/pcgen","title":"BahamutDragon/pcgen","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/BahamutDragon/pcgen"}},"updates":{"snippets":[{"icon":"PERSON","message":"@BahamutDragon in #588: I think you are confusing terminology. You have Commits, not branches, within a single branch. Each commit may be a separate issue, but it is packaged in the same branch. A branch is meant to be a focus point for a single set of changes that are related. To simplify the concept, a branch is a box where you store similar items. Barbarian Alternative Class Features would go in a box, adding aspects would be another branch/box, adding an eqmod to handle consumables would be another branch/box. All of the boxes are separate from each other. Then each box is submitted alone for review, and is accepted or rejected based upon the content of the box. When you do mass PRs, such as this one, you have a big box with completely unrelated things jumbled in. Like clothes, toys, and tools all thrown together.  On top of that, you really strain my ability to do a proper review. Every other file has a problem, but after 12 or 14 files, it becomes harder to give a good review. Especially when the changes are a jumble and inconsistent.\r\n\r\nTaken individually, these changes are positive things I would like to include. They need to be cleaned up, and they need to be separated for an easier review. "}],"action":{"name":"View Pull Request","url":"https://github.com/BahamutDragon/pcgen/pull/588#issuecomment-253704623"}}}

FerretDave commented:
Greetings,
I see 9 different branches in the github gui - as I'd expect for how I
branched/committed, so why the PR has combined them all I don't know.
However, the first one has one commit, the second has two etc, so these
have branched off each other in series (which actually sounds ok), though I
would have thought that they would shown as seperate branches to you?

I've raised a new PR - for just the first branch, with one commit.
If that's ok, then I could try raising a PR for the second branch, and
we'll see if that shows just one (more) commit at that point ?

Cheers
Dave

On 14 October 2016 at 08:13, Dave Griffin <ferret.griffin@gmail.com> wrote:

> Now I'm tearing my hair out!
>
> I *did* each change as a separate branch!!!
> ‎As per my git command notes, I created a branch, checked it out, and ‎committed
> a few files (with related changes) to it. Did a git push and repeat.
>
> Then I opened the PR - which I thought covered all of the branches i'd
> submitted... this PR should have a doszen or so branches, each commit was
> *aslo* a seperate branch... i'll take another look.
>
> Cheers
> Dave
>      oxxxxxxx{;;;;;;;;;;;;;;;;;;;;;;;;;;;;;>
> *From: *Bahamut
> *Sent: *Friday, 14 October 2016 05:04
> *To: *BahamutDragon/pcgen
> *Reply To: *BahamutDragon/pcgen
> *Cc: *Dave; Author
> *Subject: *Re: [BahamutDragon/pcgen] Backlog of fixes (#588)
>
> I think you are confusing terminology. You have Commits, not branches,
> within a single branch. Each commit may be a separate issue, but it is
> packaged in the same branch. A branch is meant to be a focus point for a
> single set of changes that are related. To simplify the concept, a branch
> is a box where you store similar items. Barbarian Alternative Class
> Features would go in a box, adding aspects would be another branch/box,
> adding an eqmod to handle consumables would be another branch/box. All of
> the boxes are separate from each other. Then each box is submitted alone
> for review, and is accepted or rejected based upon the content of the box.
> When you do mass PRs, such as this one, you have a big box with completely
> unrelated things jumbled in. Like clothes, toys, and tools all thrown
> together. On top of that, you really strain my ability to do a proper
> review. Every other file has a problem, but after 12 or 14 files, it
> becomes harder to give a good review. Especially when the changes are a
> jumble and inconsistent.
>
> Taken individually, these changes are positive things I would like to
> include. They need to be cleaned up, and they need to be separated for an
> easier review.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BahamutDragon/pcgen/pull/588#issuecomment-253704623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGNF3RWwWbP0nvUmezliquxq61Eusz0iks5qzv9PgaJpZM4KVuQP>
> .
>
>

LegacyKing commented:
Hi Dave, 

I can give you a little insight to your problem. You're cascading your individual branch work into new branches, essentially keeping them all together. It doesn't matter that you commit them in branch x, when you then created branch y, all of branch x was included. This process continued, so instead of working scope branches, you have a single branch.  

You need to branch from a single safe point for each branch to keep out "cross-contamination". Each Pull Request should be based from a single branch. If you look at the main PCGen Pull Request section, you will see that each of our PRs consist of a single branch for review and allows for finer control of what is included. 

When I make a new branch, I do so from my master branch - this is the "clean" safe point. I do not merge my other branches directly into it, instead I do a PR, and then when that branch is included, I will update my master from the upstream. 

I use tortoisegit, and can tell it to create a new branch from master instead of my current checked out branch. The command line code should look something like this: git.exe checkout --no-track -b dave_test1 master --

the -b is the branch, the first name is the new branch, the second name is where I branch from.

I hope that helps you out and decreases your hair pulling episodes.

Cheers,

BahamutDragon commented:
![branch](https://cloud.githubusercontent.com/assets/4307487/19420890/dfd6c496-93a9-11e6-85f2-3c905d7338c1.JPG)

Straight line, that is not proper branching.

FerretDave commented:
ah, so I'm branching from branches, not from master... that's where things
have gone wrong.

Explanations make sense, thank you (I still have some hair left!).

So, rather than:
branch / checkout / commit / repeat

I need to ...
branch / checkout from master / commit / repeat

And is the secret to that just using another param on the checkout command?
So... to fix the issue I've got, do I do:

git branch HealingBelt
git checkout HealingBelt master
commit

git branch RunescarredBerserker
git checkout RunescarredBerserker master
commit

etc?

Cheers
Dave

On 15 October 2016 at 16:30, Andrew Maitland <notifications@github.com>
wrote:

> Hi Dave,
>
> I can give you a little insight to your problem. You're cascading your
> individual branch work into new branches, essentially keeping them all
> together. It doesn't matter that you commit them in branch x, when you then
> created branch y, all of branch x was included. This process continued, so
> instead of working scope branches, you have a single branch.
>
> You need to branch from a single safe point for each branch to keep out
> "cross-contamination". Each Pull Request should be based from a single
> branch. If you look at the main PCGen Pull Request section, you will see
> that each of our PRs consist of a single branch for review and allows for
> finer control of what is included.
>
> When I make a new branch, I do so from my master branch - this is the
> "clean" safe point. I do not merge my other branches directly into it,
> instead I do a PR, and then when that branch is included, I will update my
> master from the upstream.
>
> I use tortoisegit, and can tell it to create a new branch from master
> instead of my current checked out branch. The command line code should look
> something like this: git.exe checkout --no-track -b dave_test1 master --
>
> the -b is the branch, the first name is the new branch, the second name is
> where I branch from.
>
> I hope that helps you out and decreases your hair pulling episodes.
>
> Cheers,
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BahamutDragon/pcgen/pull/588#issuecomment-253991114>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGNF3SxCvq6MUkCx5R3UcOvzs6M_JPK1ks5q0PGdgaJpZM4KVuQP>
> .
>

FerretDave commented:
Greetings,
Just finalising the updated branching, I've now got multiple separate
branches for these changes on my local repo, just trying to get that pushed
out to update my github repo so I can raise a new PR.

To answer various queries:

In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
realms/races_of_faerun/rac_templates.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

This should be throwing an error. |Orc ~ Light Sensitivity| |Remove
Race Traits. There should not be a '| |' in there.

DAVE> fixed, that didn't give any error on load !

------------------------------

In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
realms/unapproachable_east/una_abilities.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

																																																																																																																																																																																																																																																																																			BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,2	BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,5	BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,9	BONUS:VAR|RageTimes|RunescarExtraRage	BONUS:VAR|RunescarExtraRageLVL|CL=Runescarred
Berserker	SOURCEPAGE:p.32


Why separate statements? This can be simplified with multiple if statements
inside or using the native comparison.

DAVE> the progression is increase Natural AC by 1 at levels 2,5,9 - I found
it easier to use separate statements for this and bonus the var repeatedly.
Not sure how you'd use an 'if' statement in the lst file?
------------------------------

In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
realms/unapproachable_east/una_equip.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

Why are we electing to have ASPECT comments sitting on equipment? Either
grant the ability with the ASPECT tag, swap out for a QUALITY tag, or
figure out how to implement the INFO tag since that seems to be the
eventual direction with text output tags.

DAVE> On the PDF sheet, in the 'combat bonus' and 'save bonus' box near the
top of page 1, I want to have summary info that's useful during combat here
(my player often forgets various things on secondary pages, particularly
when OS is now nearly a dozen pages long!)

The Runescars as equipment is something I've added in myself, so shouldnt
impact anyone else adversely, but could be beneficial to some.

Having the equipment grant the ability to grant the aspect just seems a
long winded approach, but I probably will go down that route.I'll look into
INFO as an alternative.
------------------------------

In data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

Why is the ASPECT commented out?

DAVE> this is appearing on page 1 of the OS and it's a very lengthy piece
of 'summary' text, so I've shortened it to one line. I've left the full
aspect tag in as a comment, so it can be reverted if necessary, however
that full text appears in the abilities detail on page 3, so is not
required here in full. (All part of trying to make page 1 more relevant to
combat).
------------------------------

In data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

> @@ -270,7 +280,8 @@ Retain Dex to AC ~ Flat-footed												CATEGORY:Special Ability	TYPE:Internal
 ###Block: Share Uncanny Dodge Tier
 # Ability Name		Category of
Ability		Type												Description																																																																																																									Ability										Modify
VAR
 Uncanny Dodge		CATEGORY:Special Ability	TYPE:Class
Feature.SpecialQuality.Extraordinary.Uncanny Dodge					DESC:Character
retains their Dexterity bonus to AC (if any) even if caught
flat-footed or struck by an invisible attacker. However, they still
loses his Dexterity bonus to AC if
immobilized.																																																																											ABILITY:Special
Ability|AUTOMATIC|Retain Dex to AC ~ Flat-footed	VISIBLE:EXPORT
-Improved Uncanny Dodge	CATEGORY:Special
Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class
Feature.SpecialQuality.Improved Uncanny Dodge	DESC:Character can no
longer be flanked; you can react to opponents on opposite sides of you
as easily as you can react to a single attacker. This defense denies a
rogue the ability to sneak attack the character by flanking him,
unless the attacker has at least four more rogue levels than the
target has uncanny dodge levels [Rogue Level %1]. If a character
already has uncanny dodge from a second class, the character
automatically gains improved uncanny dodge instead, and the levels
from the classes that grant uncanny dodge stack to determine the
minimum level a rogue must be to flank the
character.|UncannyDodgeFlankingLevel												BONUS:VAR|UncannyDodgeFlankingLevel|4|TYPE=OnceOnly	VISIBLE:EXPORT
+Improved Uncanny Dodge	CATEGORY:Special
Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class
Feature.SpecialQuality.Improved Uncanny Dodge	DESC:You can no longer
be flanked; you can react to opponents on opposite sides of you as
easily as you can react to a single attacker. This defense denies a
rogue the ability to sneak attack you by flanking you, unless the
attacker has at least four more rogue levels than you have uncanny
dodge levels [Rogue Level exceeds
%1].|3+UncannyDodgeFlankingLevel												BONUS:VAR|UncannyDodgeFlankingLevel|4|TYPE=OnceOnly	VISIBLE:EXPORT

Why are we setting the numbers here incorrectly?

DAVE> I've only changed the description for clarity, it displays the level
the attacker must have, I'm not altering the variable or setting a value.
------------------------------

In data/35e/wizards_of_the_coast/core/players_handbook/ph_equip.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

> @@ -53,11 +53,11 @@ Touch Attack (Ray)								PROFICIENCY:WEAPON|Spells (Ray)			TYPE:Weapon.Ranged.U

 # Equipment Name			Unique Key					Required Weapon
Proficiency				Type																	Alt
Type						Contains													Cost		Weight	Slot Needed	Base
Quantity	Crit Mult	Crit Range	Alt Crit Mult	Alt Crit Range	Damage	Alt
Damage		Modifier			Alt EQModifier	Wield Category	Requires
Modification?	Range		Reach		Reach Multiplier	Size		Source
Page				Bonus Equip Mods																		Weapon prop.
bonus																									Special Property
 Arrow																			TYPE:Ammunition.Standard.Arrow.Individual																																COST:0.05	WT:0.15			BASEQTY:1																	EQMOD:STEEL.WOOD								MODS:YES												SOURCEPAGE:p.114																																																			SPROP:An
arrow used as a melee weapon is treated as a light improvised weapon
(-4 penalty on attack rolls) and deals damage as a dagger of it's size
(critical multiplier x2). Arrows come in a leather quiver that holds
20 arrows. An arrow that hit's its target is destroyed; one that
misses has a 50%% chance of being destroyed or lost.
-Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt
(Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A
crossbow bolt used as a melee weapon is treated as a light improvised
weapon (-4 penalty on attack rolls)and deals damage as a dagger of its
size (crit x2). Bolts come in a wooden case that holds 10 bolts (or 5,
for a repeating crossbow). A bolt that hits its target is destroyed;
one that misses has a 50%% chance of being destroyed or lost.
+Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt
(Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A
crossbow bolt used as a melee weapon is treated as a light improvised
weapon (-4 penalty on attack rolls)and deals damage as a dagger of its
size (crit x2). Bolts come in a wooden case that holds 10 bolts (or 5,
for a repeating crossbow). A bolt that hits its target is destroyed;
one that misses has a 50% chance of being destroyed or lost.

Why are you removing the escape character?

DAVE> because an SPROP does not substitute % chars, only DESC does ?
------------------------------

In data/35e/wizards_of_the_coast/core/universal_rules/ur_abilities.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

> @@ -917,11 +917,12 @@ Resistance to Sonic		CATEGORY:Special Ability	TYPE:ResistanceSelection	VISIBLE:N

 ###Block: Resistances Output
 # Ability Name				Output Name						

Just how many times do you need a resistance displayed? I think three times
is a bit excessive for the PDF export.

You might want to add the same functions of the pdf to the htm export for a
consistent appearance.

DAVE> this is to make it appear on page 1 of the OS, so all 'combat stats'
are in one place. 3 times? I only see them in the abilities normally. I
only use PDF, however am under the impression that ASPECT updates into
both, so output should be the same?
------------------------------

In data/35e/wizards_of_the_coast/supplement/expanded_psionics_
handbook/xph_equip.lst
<https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>:

SOURCEPAGE:16? Does not match the surrounding three digit formats

DAVE> missed the last digit, sorry, fixed.

On 14 October 2016 at 04:52, Bahamut <notifications@github.com> wrote:

> *@BahamutDragon* requested changes on this pull request.
>
> Mate, this is not a single branch for a single related change, this is
> just the last PR without the merges. I see feats, spells, alternate class
> features, etc. A branch should either be an issue, or descriptive of the
> change. Anyone reading the "RacOrcBlood' will think this change is about
> Orc Blood from the Races series. Instead this is a bulk PR covering
> everything you have changed.
> ------------------------------
>
> In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
> realms/races_of_faerun/rac_templates.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -52,4 +52,7 @@ Yuan-Ti (Tainted One)
>  #Elf ~ Moon																	VISIBLE:NO																																																																																																																																																			FAVOREDCLASS:Wizard																																LANGBONUS:Auran,Chondathan,Gnoll,Gnome,Halfling,Illuskan,Sylvan						SUBRACE:YES
>
>
> -Reincarnated into Mountain Orc	KEY:Reincarnated ~ Mountain Orc	VISIBLE:EXPORT	TYPE:Reincarnation	SIZE:M	BONUS:STAT|STR|4|TYPE=Racial	BONUS:VAR|DarkvisionRange|60|TYPE=Base	ABILITY:Special Ability|AUTOMATIC|Orc ~ Light Sensitivity|Orc ~ Orc Blood|Remove Race Traits	MOVE:Walk,30
> +# Orc Blood doesnt exist,only Half-Orc blood !
> +#Orc ~ Orc Blood
> +Reincarnated into Mountain Orc	KEY:Reincarnated ~ Mountain Orc	VISIBLE:EXPORT	TYPE:Reincarnation	SIZE:M	BONUS:STAT|STR|4|TYPE=Racial	BONUS:VAR|DarkvisionRange|60|TYPE=Base	ABILITY:Special Ability|AUTOMATIC|Orc ~ Light Sensitivity| |Remove Race Traits	MOVE:Walk,30
>
> This should be throwing an error. |Orc ~ Light Sensitivity| |Remove Race
> Traits. There should not be a '| |' in there.
> ------------------------------
>
> In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
> realms/unapproachable_east/una_abilities.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -39,7 +39,8 @@ Runescarring										CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary
>
>  Scribe Runescar		KEY:Runescarred Berserker ~ Scribe Runescar	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.SpellLike																		DESC:At 1st level, a runescarred berserker learns how to carve a runescar into her own skin. A runescar is a means of storing a spell, much like a scroll. The berserker knows only a limited number of spells with which to imbue runescars, selected from the runescarred berserker spell list below. To learn a runescar spell, the berserker must have a Wisdom score of at least 10 + the spell level. A runescar berserker can only cast spells by crafting them as runescars; she has no other spellcasting capability and cannot use spell completion or spell trigger magic items based on spells from this list. Scribing a runescar takes 1 hour, the expenditure of materials equal to 5 gp x spell level x caster level, and the expenditure of XP equal to 1/25th the gp spent. The character may set the caster level of the runescar at anywhere from the minimum class level necessary for her to know the rune to her own runescarred berserker class level. She may scribe a runescar only on her own body. The act of scribing a runescar is painful and deals 1d6 points of damage per spell level at the end of the scribing time. (The berserker is free to heal this damage by any normal means afterward-the runescar remains until activated.) A runescarred berserker can bear no more than seven runescars at any one time: two on the chest, two on the upper arms, two on the backs of the hands, and one on the face. Runescars are dormant until activated and cannot be dispelled (although they can be suppressed). Runescars are considered divine spells, although a berserker does not actively prepare or cast them. The save DC, if necessary, is 10 + spell level + the berserker's Wisdom score. To activate a runescar, the character must take a standard action and touch the rune. All runescar spells target only the runescarred berserker or an item in her possession, even if the spell scribed could normally be bestowed on another subject. After activation, the runescar's magic is discharged, and the rune fades to nothing but a faint outline. Typically, a berserker's skin is crisscrossed with dozens of old, discharged runescars.																																																			SOURCEPAGE:p.32
>  Extra Rage			KEY:Runescarred Berserker ~ Extra Rage		CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary			DEFINE:RageTimes|0	DEFINE:RunescarExtraRage|0	DEFINE:RunescarExtraRageLVL|0	DESC:A runescarred berserker gains additional daily uses of the barbarian rage ability. At 2nd level, she gains one additional rage per day; at 5th level, she gains a second additional rage; and at 9th level, she gains a third additional rage. For example, a barbarian 8/runescarred berserker 5 can rage three times per day from her barbarian levels, plus two more times from her runescarred berserker levels, for a total of five rages per day.																																																																																																																																																																																																																																																																																							BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,2	BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,5	BONUS:VAR|RunescarExtraRage|1|PREVARGTEQ:RunescarExtraRageLVL,9	BONUS:VAR|RageTimes|RunescarExtraRage	BONUS:VAR|RunescarExtraRageLVL|CL=Runescarred Berserker	SOURCEPAGE:p.32
> -Ritual Scarring		KEY:Runescarred Berserker ~ Ritual Scarring	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary																	DESC:Through frequent disfiguration of her own skin, a runescarred berserker gains a +1 natural armor bonus to AC at 3rd level. This increases to +2 at 6th level, and +3 at 9th level.																																																																																																																																																																																																																																																																																																																																																																																					SOURCEPAGE:p.32
> -Spawn of the Frost	KEY:Runescarred Berserker ~ Spawn of the Frost	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Supernatural																	DESC:Inured to the bitter cold of Rashemen's winters, a runescarred berserker gains cold resistance 5.																																																																																																																																																																																																																																																																																																																																																																																																		SOURCEPAGE:p.32
> +Ritual Scarring		KEY:Runescarred Berserker ~ Ritual Scarring	CATEGORY:Special Ability	TYPE:Runescarred Berserker Class Feature.SpecialQuality.Extraordinary																	DESC:Through frequent disfiguration of her own skin, a runescarred berserker gains a +1 natural armor bonus to AC at 3rd level. This increases to +2 at 6th level, and +3 at 9th level.																																																																																																																																																																																																																																																																																																																																																																																					DEFINE:RunescarNaturalAC|0	BONUS:VAR|RunescarNaturalAC|1|PREVARGTEQ:RunescarredBerserkerLVL,3	BONUS:VAR|RunescarNaturalAC|1|PREVARGTEQ:RunescarredBerserkerLVL,6	BONUS:VAR|RunescarNaturalAC|1|PREVARGTEQ:RunescarredBerserkerLVL,9	BONUS:COMBAT|AC|RunescarNaturalAC|TYPE=NaturalArmor	SOURCEPAGE:p.32
>
> Why separate statements? This can be simplified with multiple if
> statements inside or using the native comparison.
> ------------------------------
>
> In data/35e/wizards_of_the_coast/campaign_settings/forgotten_
> realms/unapproachable_east/una_equip.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -57,30 +57,48 @@ Silver Sign	TYPE:Magic.Wondrous.Sign.Consumable	MODS:REQUIRED		COST:5000	WT:0.01
>  # Runescars for Runescarred Berserker
>  # 1st level
>  Runescar of Cure Moderate Wounds	SPROP:Cure 2d8+ % HP|(min((classlevel("Runescarred Berserker")),30))	TYPE:Magic.Consumable	PRESPELL:1,Cure Moderate Wounds	PRECLASS:1,Runescarred Berserker=1
> +#DG todo - cannot add ASPECT to equipment, JIRA'd, may workaround by having equipment add an ability that then adds the aspect...
> +#ASPECT:CombatBonus|+% luck bonus to hit/damage for 1 minute (Runescar of Divine Favour)|min(min((classlevel("Runescarred Berserker"))/3,1),3)
>
> Why are we electing to have ASPECT comments sitting on equipment? Either
> grant the ability with the ASPECT tag, swap out for a QUALITY tag, or
> figure out how to implement the INFO tag since that seems to be the
> eventual direction with text output tags.
> ------------------------------
>
> In data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> >
>  ###Block: Dwarf - Abilities
>  # Ability Name		Unique Key					Category of Ability		Type											Visible		Define					Description																																																																																																																		Bonus to Situation											Bonus to skill																						Modify VAR											Weapon prof. bonus						Change Weapon Prof. Category					Source Page		Aspects																																																																																																															Temporary Bonus
>  Darkvision			KEY:Dwarf ~ Darkvision			CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.RacialVision.DwarfDarkvision	VISIBLE:DISPLAY							DESC:Dwarves can see in the dark up to %1 feet. Darkvision is black and white only, but it is otherwise like normal sight, and dwarves can function just fine with no light at all.|DarkvisionRange																																																																																																																									BONUS:VAR|RacialVision|1	BONUS:VAR|DarkvisionRange|60|TYPE=Base																			SOURCEPAGE:p.14
>  Weapon Familiarity	KEY:Dwarf ~ Weapon Familiarity	CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY							DESC:Dwarves may treat dwarven waraxes and dwarven urgroshes as martial weapons, rather than exotic weapons.																																																																																																																																																			BONUS:WEAPONPROF=Waraxe (Dwarven)|WIELDCATEGORY|-1	CHANGEPROF:Urgrosh (Dwarven),Waraxe (Dwarven)=Martial	SOURCEPAGE:p.15
> -Stonecunning		KEY:Dwarf ~ Stonecunning		CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY	DEFINE:StonecunningSearchBonus|0	DESC:This ability grants a dwarf a +%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings, and the like. Something that isn't stone but that is disguised as stone also counts as unusual stonework. A dwarf who merely comes within 10 feet of unusual stonework can make a Search check as if he were actively searching, and a dwarf can use the Search skill to find stonework traps as a rogue can. A dwarf can also intuit depth, sensing his approximate depth underground as naturally as a human can sense which way is up.|StonecunningSearchBonus	BONUS:SITUATION|Search=Notice unusual stonework|StonecunningSearchBonus|TYPE=Racial																									BONUS:VAR|StonecunningSearchBonus|2|TYPE=Racial																							SOURCEPAGE:p.15	ASPECT:SkillBonus|+%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings, and the like. Something that isn't stone but that is disguised as stone also counts as unusual stonework. A dwarf who merely comes within 10 feet of unusual stonework can make a Search check as if he were actively searching, and a dwarf can use the Search skill to find stonework traps as a rogue can. A dwarf can also intuit depth, sensing his approximate depth underground as naturally as a human can sense which way is up.|StonecunningSearchBonus
> -Stability			KEY:Dwarf ~ Stability			CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY							DESC:+4 bonus on ability checks made to resist being bull rushed or tripped when standing on the ground (but not when climbing, flying, riding, or otherwise not standing firmly on the ground).																																																																																																																																																							SOURCEPAGE:p.15	ASPECT:CombatBonus|+4 bonus on ability checks made to resist being bull rushed or tripped when standing on the ground (but not when climbing, flying, riding, or otherwise not standing firmly on the ground).
> +Stonecunning		KEY:Dwarf ~ Stonecunning		CATEGORY:Special Ability	TYPE:Race Trait.Dwarf Race Trait.SpecialQuality				VISIBLE:DISPLAY	DEFINE:StonecunningSearchBonus|0	DESC:This ability grants a dwarf a +%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings, and the like. Something that isn't stone but that is disguised as stone also counts as unusual stonework. A dwarf who merely comes within 10 feet of unusual stonework can make a Search check as if he were actively searching, and a dwarf can use the Search skill to find stonework traps as a rogue can. A dwarf can also intuit depth, sensing his approximate depth underground as naturally as a human can sense which way is up.|StonecunningSearchBonus	BONUS:SITUATION|Search=Notice unusual stonework|StonecunningSearchBonus|TYPE=Racial																									BONUS:VAR|StonecunningSearchBonus|2|TYPE=Racial																							SOURCEPAGE:p.15	ASPECT:SkillBonus|+%1 racial bonus on Search checks to notice unusual stonework.|StonecunningSearchBonus
> +#ASPECT:SkillBonus|+%1 racial bonus on Search checks to notice unusual stonework, such as sliding walls, stonework traps, new construction (even when built to match the old), unsafe stone surfaces, shaky stone ceilings, and the like. Something that isn't stone but that is disguised as stone also counts as unusual stonework. A dwarf who merely comes within 10 feet of unusual stonework can make a Search check as if he were actively searching, and a dwarf can use the Search skill to find stonework traps as a rogue can. A dwarf can also intuit depth, sensing his approximate depth underground as naturally as a human can sense which way is up.|StonecunningSearchBonus
>
> Why is the ASPECT commented out?
> ------------------------------
>
> In data/35e/wizards_of_the_coast/core/players_handbook/ph_abilities.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -270,7 +280,8 @@ Retain Dex to AC ~ Flat-footed												CATEGORY:Special Ability	TYPE:Internal
>  ###Block: Share Uncanny Dodge Tier
>  # Ability Name		Category of Ability		Type												Description																																																																																																									Ability										Modify VAR
>  Uncanny Dodge		CATEGORY:Special Ability	TYPE:Class Feature.SpecialQuality.Extraordinary.Uncanny Dodge					DESC:Character retains their Dexterity bonus to AC (if any) even if caught flat-footed or struck by an invisible attacker. However, they still loses his Dexterity bonus to AC if immobilized.																																																																											ABILITY:Special Ability|AUTOMATIC|Retain Dex to AC ~ Flat-footed	VISIBLE:EXPORT
> -Improved Uncanny Dodge	CATEGORY:Special Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class Feature.SpecialQuality.Improved Uncanny Dodge	DESC:Character can no longer be flanked; you can react to opponents on opposite sides of you as easily as you can react to a single attacker. This defense denies a rogue the ability to sneak attack the character by flanking him, unless the attacker has at least four more rogue levels than the target has uncanny dodge levels [Rogue Level %1]. If a character already has uncanny dodge from a second class, the character automatically gains improved uncanny dodge instead, and the levels from the classes that grant uncanny dodge stack to determine the minimum level a rogue must be to flank the character.|UncannyDodgeFlankingLevel												BONUS:VAR|UncannyDodgeFlankingLevel|4|TYPE=OnceOnly	VISIBLE:EXPORT
> +Improved Uncanny Dodge	CATEGORY:Special Ability	TYPE:Extraordinary.ImprovedUncannyDodge.Class Feature.SpecialQuality.Improved Uncanny Dodge	DESC:You can no longer be flanked; you can react to opponents on opposite sides of you as easily as you can react to a single attacker. This defense denies a rogue the ability to sneak attack you by flanking you, unless the attacker has at least four more rogue levels than you have uncanny dodge levels [Rogue Level exceeds %1].|3+UncannyDodgeFlankingLevel												BONUS:VAR|UncannyDodgeFlankingLevel|4|TYPE=OnceOnly	VISIBLE:EXPORT
>
> Why are we setting the numbers here incorrectly?
> ------------------------------
>
> In data/35e/wizards_of_the_coast/core/players_handbook/ph_equip.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -53,11 +53,11 @@ Touch Attack (Ray)								PROFICIENCY:WEAPON|Spells (Ray)			TYPE:Weapon.Ranged.U
>
>  # Equipment Name			Unique Key					Required Weapon Proficiency				Type																	Alt Type						Contains													Cost		Weight	Slot Needed	Base Quantity	Crit Mult	Crit Range	Alt Crit Mult	Alt Crit Range	Damage	Alt Damage		Modifier			Alt EQModifier	Wield Category	Requires Modification?	Range		Reach		Reach Multiplier	Size		Source Page				Bonus Equip Mods																		Weapon prop. bonus																									Special Property
>  Arrow																			TYPE:Ammunition.Standard.Arrow.Individual																																COST:0.05	WT:0.15			BASEQTY:1																	EQMOD:STEEL.WOOD								MODS:YES												SOURCEPAGE:p.114																																																			SPROP:An arrow used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls) and deals damage as a dagger of it's size (critical multiplier x2). Arrows come in a leather quiver that holds 20 arrows. An arrow that hit's its target is destroyed; one that misses has a 50%% chance of being destroyed or lost.
> -Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt (Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A crossbow bolt used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls)and deals damage as a dagger of its size (crit x2). Bolts come in a wooden case that holds 10 bolts (or 5, for a repeating crossbow). A bolt that hits its target is destroyed; one that misses has a 50%% chance of being destroyed or lost.
> +Bolt					TYPE:Ammunition.Standard.Bolt.Individual																																KEY:Bolt (Crossbow)																																																																																																SOURCEPAGE:p.115																																																			SPROP:A crossbow bolt used as a melee weapon is treated as a light improvised weapon (-4 penalty on attack rolls)and deals damage as a dagger of its size (crit x2). Bolts come in a wooden case that holds 10 bolts (or 5, for a repeating crossbow). A bolt that hits its target is destroyed; one that misses has a 50% chance of being destroyed or lost.
>
> Why are you removing the escape character?
> ------------------------------
>
> In data/35e/wizards_of_the_coast/core/universal_rules/ur_abilities.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> > @@ -917,11 +917,12 @@ Resistance to Sonic		CATEGORY:Special Ability	TYPE:ResistanceSelection	VISIBLE:N
>
>  ###Block: Resistances Output
>  # Ability Name				Output Name						Category of Ability		Type								Visible	Description																			Aspects
> -Resistance to Acid Output		OUTPUTNAME:Resistance to Acid			CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Acid damage each time you take Acid damage|AcidResistanceBonus					ASPECT:Resistance|Acid	ASPECT:Ability Benefit|%1|AcidResistanceBonus
> -Resistance to Cold Output		OUTPUTNAME:Resistance to Cold			CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Cold damage each time you take Cold damage|ColdResistanceBonus					ASPECT:Resistance|Cold	ASPECT:Ability Benefit|%1|ColdResistanceBonus
> -Resistance to Fire Output		OUTPUTNAME:Resistance to Fire			CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Fire damage each time you take Fire damage|FireResistanceBonus					ASPECT:Resistance|Fire	ASPECT:Ability Benefit|%1|FireResistanceBonus
> -Resistance to Electricity Output	OUTPUTNAME:Resistance to Electricity	CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Electricity damage each time you take Electricity damage|ElectricityResistanceBonus	ASPECT:Resistance|Electricity	ASPECT:Ability Benefit|%1|ElectricityResistanceBonus
> -Resistance to Sonic Output		OUTPUTNAME:Resistance to Sonic		CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Sonic damage each time you take Sonic damage|SonicResistanceBonus				ASPECT:Resistance|Sonic	ASPECT:Ability Benefit|%1|SonicResistanceBonus
> +Resistance to Acid Output		OUTPUTNAME:Resistance to Acid			CATEGORY:Special Ability	TYPE:SpecialQuality.Extraordinary.Resistance	VISIBLE:YES	DESC:You may ignore %1 points of Acid damage each time you take Acid damage|AcidResistanceBonus					ASPECT:SaveBonus|Acid Resistance %1|AcidResistanceBonus	ASPECT:Resistance|Acid	ASPECT:Ability Benefit|%1|AcidResistanceBonus
> +# DG - not seeing the original ASPECT features output anywhere, so adding additional
>
> Just how many times do you need a resistance displayed? I think three
> times is a bit excessive for the PDF export.
>
> You might want to add the same functions of the pdf to the htm export for
> a consistent appearance.
> ------------------------------
>
> In data/35e/wizards_of_the_coast/supplement/expanded_psionics_
> handbook/xph_equip.lst
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>
> :
>
> >
>  # Psicrown
> -Psicrown (Astral Legion)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:47250	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 900; Manifester Level - 18; Powers - Astral Construct
> -Psicrown (Beast)									OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:33750	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 450; Manifester Level - 9; Powers - Claw of Energy, Duodimensional Claw, Prevenom, Psionic Lion's Charge, Truevenom
> -Psicrown (Cautious Warrior)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:32063	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 450; Manifester Level - 9; Powers - Adapt Body, Body Adjustment, Precognition (Defensive), Inertial Barrior
> -Psicrown (Discerning Watcher)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:51469	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 450; Manifester Level - 9; Powers - Aura Sight, Clairtangent Hand, Clairvoyant Sense, Darkvision (Psionic), Remote Viweing, True Seeing (Psionic)
> -Psicrown (Dominator)								OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:20250	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 450; Manifester Level - 9; Powers - Charm (Psionic), Dominate (Psionic), Suggestion (Psionic)
> -Psicrown (Evader)									OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:28500	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 400; Manifester Level - 8; Powers - Catfall, Hustle, Fly (Psionic), Freedom of Movement (Psionic), Wall Walker
> -Psicrown (Fiery Ruin)								OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:67500	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 750; Manifester Level - 15; Powers - Energy Ball, Energy Cone, Energy Missile, Energy Stun, Fiery Discorporation
> -Psicrown (Great Dominator)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:45000	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 750; Manifester Level - 15; Powers - Charm (Psionic), Dominate (Psionic), Suggestion (Psionic)
> -Psicrown (Temporal Juggler)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:95625	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 850; Manifester Level - 17; Powers - Temporal Acceleration, Time Hop, Timeless Body
> -Psicrown (Traveler)								OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:80156	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:PsionicItems.rtf																																																																						SPROP:Power Points - 750; Manifester Level - 15; Powers - Astral Traveler, Dimension Door (Psionic), Teleport (Greater (Psionic)), Teleport (Psionic)
> +Psicrown (Astral Legion)							OUTPUTNAME:Psicrown of the [NAME]								TYPE:Psionic.Psicrown						COST:47250	WT:0				BASEITEM:Psicrown																							SOURCEPAGE:16																																																																						SPROP:Power Points - 900; Manifester Level - 18; Powers - Astral Construct
>
> SOURCEPAGE:16? Does not match the surrounding three digit formats
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BahamutDragon/pcgen/pull/588#pullrequestreview-4202645>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGNF3XCNmg2N_WzKzSegcHJ4vvuFJMmvks5qzvyBgaJpZM4KVuQP>
> .
>
